src/libcode/vx_tc_util/track_info.cc

```diff
@@ -1138,6 +1162,52 @@ TrackInfo consensus(const TrackInfoArray &tracks,
       // Compute consensus CycloneLevel
       if(!is_bad_data(pavg.v_max())) pavg.set_level(wind_speed_to_cyclonelevel(pavg.v_max()));
 
+      // Loop over the diag name and the input track points and get consensus diag value
+      // (mean value across all members that have the diag value)
+      for (j=0; j<tavg.diag_name().n(); j++){
+         mlog << Debug(3) << "TrackInfo::consensus(). Working on calculating mean values for tavg.diag_name()[" << j << "] = " << tavg.diag_name()[j] << "\n";
+         
+         // Store diag_vals for one diag_name across all lead-times (track points)
+         NumArray diag_vals;
+         double cons_diag_val;
+
+         // Initializ diag_vals array for this diag name
+         diag_vals.clear();
+
+         // Loop through the tracks to get average diagnostic values
+         for(k=0; k<tracks.n(); k++) {
+
+            // Get the index of the TrackPoint for this lead time
+            i_pnt = tracks.Track[k].lead_index(nint(lead_list[i]));
+
+            // Check for missing TrackPoint in a required member
+            if(i_pnt < 0) {
+               if(req_list.has(tracks.Track[k].technique())) {
+                  skip = true;
+                  break;
+               }
+               continue;
+            }
+
+            // Add non-missing diag values to local NumArray 
+            if(!is_bad_data( tracks.Track[k][i_pnt].get_diag_val(tavg.diag_name(), tavg.diag_name()[j]) )) {
+               mlog << Debug(3) <<  "  Adding values for k: " << k << ", i_pnt: " << i_pnt << " tracks.Track[k][i_pnt].get_diag_val(tavg.diag_name(), tavg.diag_name()[j]) = " << tracks.Track[k][i_pnt].get_diag_val(tavg.diag_name(), tavg.diag_name()[j]) << "\n";
+               diag_vals.add(tracks.Track[k][i_pnt].get_diag_val(tavg.diag_name(), tavg.diag_name()[j]));
+            }
+            
+         } // end loop over ensemble of tracks (for given lead-time: i_pnt)
+         
+         cons_diag_val = diag_vals.mean();
+         mlog << Debug(3) << "  cons_diag_val = " << cons_diag_val << "\n";
+         
+         // Add the cons_diag_val to the pavg DiagVal NumArray (using add_diag_value)
+         if(!is_bad_data(cons_diag_val)) {
```

Comment:
Don't we actually NEED to store the computed diagnostic value regardless of whether or not it contains bad data? For example, let's say all the input tracks report a diag value with bad data. If we fail to write bad data to the output, we'll end up with a mismatch of diagnostic names and values.

Reply:
Hi John, I wasn't sure if we wanted to save the missing data. But based on what you are saying it makes sense.